validate-pr/validate_pr.sh

```diff
@@ -0,0 +1,91 @@
+RFB_PATTERN='^([0-9]+\.[0-9]+)_fb_.+'             # 20.11_fb_backportFeature_12345
+
+{
+    # 'fb_*' should target 'develop'
+    if [[ "$HEAD_BRANCH" =~ $FB_PATTERN ]]; then
+        if [ "$BASE_BRANCH" != "develop" ]; then
+            echo "Expect PR from \`${HEAD_BRANCH}\` to target \`develop\`, not \`${BASE_BRANCH}\`"
+            if [[ "$BASE_BRANCH" =~ $SNAP_PATTERN ]] || [[ "$BASE_BRANCH" =~ $REL_PATTERN ]]; then
+                version=${BASH_REMATCH[1]}
+                echo "If this branch is intended for ${version}, it should be named \`${version}_${HEAD_BRANCH}\`"
+                echo "_Note: A new PR will have to be created_"
+            fi
+        fi
+    # 'XX.Y_fb_*' should target 'releaseXX.Y-SNAPSHOT'
+    elif [[ "$HEAD_BRANCH" =~ $RFB_PATTERN ]]; then
+        version=${BASH_REMATCH[1]}
+        expected_branch="release${version}-SNAPSHOT"
+        if [ "$BASE_BRANCH" != "$expected_branch" ]; then
+            echo "Expect PR from \`${HEAD_BRANCH}\` to target \`${expected_branch}\`, not \`${BASE_BRANCH}\`"
+        fi
+    # 'ff_XX.Y.Z' should target 'releaseXX.Y'
+    elif [[ "$HEAD_BRANCH" =~ $FF_PATTERN ]]; then
+        version=${BASH_REMATCH[1]}
+        expected_branch="release${version}"
+        if [ "$BASE_BRANCH" != "$expected_branch" ]; then
+            echo "Expect PR from \`${HEAD_BRANCH}\` to target \`${expected_branch}\`, not \`${BASE_BRANCH}\`"
+            echo "_Note: \`ff_*\` branches are reserved for LabKey's semi-automated release process._"
+        fi
+    # Warn about non-standard branch targeting a release branch
+    elif [[ "$BASE_BRANCH" =~ $SNAP_PATTERN ]] || [[ "$BASE_BRANCH" =~ $REL_PATTERN ]]; then
+        version=${BASH_REMATCH[1]}
+        echo "Branch doesn't match LabKey naming scheme"
+        echo "Branch intended for \`${version}\` should be named something like \`${version}_fb_${HEAD_BRANCH}\`"
+        echo "_Note: A new PR will have to be created_"
+    # Warn about non-standard branch targeting 'develop'
+    elif [ "$BASE_BRANCH" == "develop" ]; then
+        echo "Branch doesn't match LabKey naming scheme"
+        echo "Branch intended for \`develop\` should be named something like \`fb_${HEAD_BRANCH}\`"
+        echo "_Note: A new PR will have to be created_"
+    fi
+
+    # Only 'ff_*' branches should target non-snapshot release branch
+    if [[ "$BASE_BRANCH" =~ $REL_PATTERN ]] && ! [[ "$HEAD_BRANCH" =~ $FF_PATTERN ]]; then
+        echo "_Note: Pull requests should not target non-snapshot release branches directly._"
+    fi
+} >>"$ERROR_FILE"
+
+DEFAULT_TITLE="^Fb | fb "
+# Warn about using default pull request title
+if [[ "$PR_TITLE" =~ $DEFAULT_TITLE ]]; then
+    if [ "$(wc -l <"${ERROR_FILE}")" -gt 0 ]; then
+        echo "" >>"$ERROR_FILE" # Insert blank line after other errors.
+    fi
+    echo "This PR appears to have the default title generated by GitHub. Please use something more descriptive." >>"$ERROR_FILE"
+fi
```

Comment:
Seems like this check could occur at the start of the script-- although I don't see any `git` or `hub` use in the earlier logic, so moving this probably won't make for time savings.